src/compile/mark/mark.ts

```diff
@@ -239,8 +239,8 @@ function getStackByFields(model: UnitModel) {
         const fieldDef: FieldDef = channelEncoding;
         const scale = model.scale(channel);
         fields.push(field(fieldDef, {
-          binSuffix: scale && scale.type === ScaleType.ORDINAL ? '_range' : '_start'
-        }));
+          binSuffix: scale && scale.type === ScaleType.ORDINAL ? 'range' : 'start'
+        }) || 'undefined');
```

Comment:
Agreed, but felt this was out of scope because I don't really _get_ groupby and felt whomever implemented it would be better to address the inclusion. 

Similarly, I extended a bad pattern in timeunit.ts, with the RegExp. I started to take a crack at fixing it, but didn't think it made sense within this PR. Happy to create news issues for these if you agree.

Those seem like more 'nice to have' and less 'let's get this feature done ASAP' -- up to you!